app/components/reporting/reporting-line-items-edit/reporting-line-items-edit-directive.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+angular.module('lineItems.edit.directive', [
+  'apiServices.lineItem.factory',
+  'bulbsCmsApp.settings',
+  'lodash',
+  'saveButton.directive',
+  'topBar'
+])
+  .directive('lineItemsEdit', function (routes) {
+    return {
+      controller: function (_, $location, $q, $routeParams, $scope, LineItem) {
+        if ($routeParams.id === 'new') {
+          $scope.model = LineItem.$build();
+          $scope.isNew = true;
+        } else {
+          $scope.model = LineItem.$find($routeParams.id);
+        }
+
+        window.onbeforeunload = function (e) {
+          if (!_.isEmpty($scope.model.$dirty()) || $scope.isNew || $scope.needsSave) {
```

Comment:
Use jquery here and do ```$(window).on('beforeunload', function (){ ... })``` instead.

With the current code you run the risk of overwriting what some other module may have set to ```window.onbeforeunload``` and/or some other module could overwrite your code.